module_generator/README.md

```diff
@@ -0,0 +1,220 @@
+- `model_name` (str): Set the name of the model of the resource.
+- `initialize_workflow` (bool): Initialize a workflow for building the module and other cloud build info.
+
+## Template Structure
+
+When a module is created using the template, there are a couple of different files created or populated. Some files to note:
+
+- `.viam-gen-info` saves the user inputs, the module creation time, and the generator version.
+- `meta.json` saves the `module_name`, `visibility` (default to public), `language`, `generator_version`, `model`, `api`, and `entrypoint` (default to `run.sh`) into a json file.
+- `src/__main__.py` contains all the functions available to the resource.
+
+---
+
+## Module Creation Example
+
+Create a Python module named `rover` in the namespace `test`. Add a vision service model `vizz`. Do not initialize a github workflow.
+
+```python
+% python3 cookiecutter-template
+  [1/6] module_name (MyModule): rover
+  [2/6] Select language
+    1 - python
+    Choose from [1] (1): 1
+  [3/6] namespace (ns): test
+  [4/6] Select resource
+    1 - arm component
+    2 - base component
+    3 - motor component
+    4 - vision service
+    Choose from [1/2/3/4] (1): 4
+  [5/6] model_name (model): vizz
+  [6/6] initialize_workflow (n): n
+```
+
+### Generated Files
+
+`module_generator/rover/.viam-gen-info`
+
+```json
+{
+  "module_name": "rover",
+  "language": "python",
+  "namespace": "test",
+  "resource": "vision service",
+  "model_name": "vizz",
+  "generator_version": "0.14.0",
+  "generated_on": "2024-08-21 11:50:11"
+}
+```
+
+`module_generator/rover/meta.json`
+
+```json
+{
+  "module_name": "rover",
+  "visibility": "public",
+  "language": "python",
+  "generator": "0.14.0",
+  "models": [
+    {
+      "api": "rdk:service:vision",
+      "model": "test:rover:vizz"
+    }
+  ],
+  "entrypoint": "run.sh"
+}
+```
+
+`module_generator/rover/README.md`
+
+```
+# rover Module
+
+This module implements the Vision API in the `test:rover:vizz` model.
+
+More information on the Vision Service can be found here: https://python.viam.dev/autoapi/viam/services/vision/index.html.
+
+```
+
+`module_generator/rover/src/__main__.py`
+
+This is the raw file right after generation. This newly created file has some missing imports, which will be fixed below.
+
+```python
+import asyncio
+from typing import ClassVar, Mapping, Optional, Self, Sequence, Union
+from viam.module.module import Module
+from viam.proto.app.robot import ComponentConfig
+from viam.proto.common import ResourceName
+from viam.resource.base import ResourceBase
+from viam.resource.easy_resource import EasyResource
+from viam.resource.types import Model, ModelFamily
+from viam.services.vision import Vision
+
+
+class Vizz(Vision, EasyResource):
+    """This is your custom Vision service.
+
+    All the methods listed are optional.
+    You can implement the methods you want and remove the methods you don't need.
```

Comment:
I'm not sure if it's great to recommend that people can just remove methods. Technically the method is _not_ optional; the abstract base class defines that these methods are what makes something a `Vision` service. Given how python works it's okay to remove the method, but if, e.g., a `stop` method was removed and someone tried to call it they would get fallback to the definition in the ABC, which is basically just `pass`. If we leave it as-is here, we'd get a `NotImplementedError`. The latter seems better to me imo, otherwise it would be easy for someone to acting in reliance on a method that doesn't actually do anything.

Reply:
This makes sense. This was copied and pasted from another template, so I'll just erase the `remove the methods you don't need` part.